File: calculator2/NullValue.php
<?php

class NullValue implements Expression{


    public function evaluate()
    {
        throw new Exception();
    }

    public function isNULL(){
        return true;
    }

    public function isBinaryOperator(){
        return false;
    }

    public function isUnaryOperator(){
        return false;
    }
}

File: calTeacherCode/NullValue.java

public class NullValue implements Expression {
	@Override
	public int evaluate() {
		throw new RuntimeException();
	}
	
	@Override
	public boolean isNull() {
		return true;
	}

	@Override
	public boolean isBinaryOperator() {
		return false;
	}

	@Override
	public boolean isUnaryOperator() {
		return false;
	}
}
